### PR TITLE
2841b annotations context menu is cut

### PR DESCRIPTION
**Pull request by ramtob**

### CODE REVIEW

File: src/app/@ansyn/ol/plugins/annotations/annotations-context-menu/components/annotations-color/annotations-color.component.ts [hunk 1]
```diff
@@ -20,43 +22,26 @@ export class AnnotationsColorComponent implements AfterViewInit, OnDestroy {
...
-		const myRect = myDiv.getBoundingClientRect();
-		const imageryRect = this.imageryElement.getBoundingClientRect() as DOMRect;
-		const delta = myRect.left - imageryRect.left + myRect.width - imageryRect.width + 3;
-		if (delta > 0) {
-			this.moveLeft += delta;
-		} else {
-			this.moveLeft = Math.max(0, this.moveLeft + delta);
-		}
+	ngOnDestroy(): void {
+		this.stayInImageryService.destroy();
```
[the remark this answers is not in the document]
Comment (asafMasa) [reply src/app/@ansyn/ol/plugins/annotations/annotations-context-menu/components/annotations-color/annotations-color.component.ts]: check for memory leaks

File: src/app/@ansyn/ol/helpers/stay-in-imagery.service.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
+import { Injectable } from '@angular/core';
```
[the remark this answers is not in the document]
Comment (ramtob) [reply src/app/@ansyn/ol/helpers/stay-in-imagery.service.ts]: done

File: src/app/@ansyn/ol/helpers/stay-in-imagery.service.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+@Injectable()
+export class StayInImageryService {
+	getElementFunc: Function;
+	timerCallback: Function;
+	timerId: number;
+	public moveLeft = 0;
+	public moveDown = 0;
+
+	init(funcOrElement: Function | Element, timerCallback: Function = null) {
```
[the remark this answers is not in the document]
Comment (ramtob) [reply src/app/@ansyn/ol/helpers/stay-in-imagery.service.ts]: done

File: src/app/@ansyn/ol/plugins/annotations/annotations-context-menu/components/annotations-color/annotations-color.component.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (ramtob) [reply src/app/@ansyn/ol/plugins/annotations/annotations-context-menu/components/annotations-color/annotations-color.component.ts]: Found memory leak - but does not seem related to my changes - will open another issue on this.

File: src/app/@ansyn/imagery/stay-in-imagery-service/stay-in-imagery.service.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+	constructor(@Inject(IMAGERY_CONFIG) protected imageryConfig: IImageryConfig) {
+	}
+
+	init(elementOrElementCallback: Function | Element, timerCallback: Function = null) {
+		this.elementCallback = elementOrElementCallback instanceof Function ? elementOrElementCallback : () => elementOrElementCallback;
+		this.timerCallback = timerCallback;
+		this.timerId = window.setInterval(this.calcPositionToStayInsideImagery.bind(this), 300);
+	}
+
+	destroy() {
```
Comment (asafMasa) [inline src/app/@ansyn/imagery/stay-in-imagery-service/stay-in-imagery.service.ts]: check if there is a build in ngOnDestroy for service

Comment (ramtob) [reply src/app/@ansyn/imagery/stay-in-imagery-service/stay-in-imagery.service.ts]: 1 Yes there is ngOnDestroy for service!
2 I added it to the Stay service, and deleted the destroy() method.

**Pull request merged**